### PR TITLE
Clean up Home Composable

### PR DESCRIPTION
**Pull request by boswelja**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

The Home Composable featured a lot of duplicate code. 

**Summary of changes:**

1. Slightly reduced MediaSmall complexity and improved consistency
2. Extracted common Composables from Home Composable

**Tested changes:**

Home screen still functions

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt [hunk 1]
```diff
@@ -135,177 +138,70 @@ fun Home(
...
                                 }
                             )
 
-                            Spacer(
-                                Modifier.height(
-                                    dimensionResource(Res.dimen.navigation_bar_height)
-                                            + dimensionResource(Res.dimen.large_padding)
-                                )
-                            )
+                            Spacer(Modifier.height(dimensionResource(Res.dimen.navigation_bar_height)))
```
Comment (imashnake0) [inline app/src/main/java/com/imashnake/animite/features/home/Home.kt:204]: Bottom padding is not the same as before.

**Review by imashnake0 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt [hunk 2]
```diff
@@ -345,3 +241,43 @@ fun Home(
...
+fun HomeRow(
+    list: List<MediaListQuery.Medium?>,
+    title: String,
+    onItemClicked: (MediaListQuery.Medium) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier) {
+        Text(
+            text = title,
+            style = MaterialTheme.typography.titleMedium,
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:255]: Color is pulled from the theme automatically, did it change?

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt (hunk 1 above)
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:204]: Did I forget to move it up a level? Oop

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt [hunk 3]
```diff
@@ -345,3 +241,43 @@ fun Home(
...
+                start = dimensionResource(Res.dimen.large_padding)
+            )
+        )
+
+        Spacer(Modifier.size(dimensionResource(Res.dimen.medium_padding)))
+
+        MediaSmallRow(
+            mediaList = list,
+            onItemClick = { itemId ->
+                if (itemId != null) { onItemClicked(list.first { it?.id == itemId }!!) }
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:266]: Welcome to "everything is nullable", I hope you enjoy your stay

File: app/src/main/java/com/imashnake/animite/features/ui/MediaSmall.kt [hunk 4]
```diff
@@ -53,7 +52,7 @@ fun <T> MediaSmallRow(
             end = dimensionResource(Res.dimen.large_padding)
         )
     ) {
-        items(mediaList.filterNotNull()) { media ->
+        items(mediaList) { media ->
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/ui/MediaSmall.kt:55]: There are no nullable elements (that I can tell), everything is just nullable

File: app/src/main/java/com/imashnake/animite/features/ui/MediaSmall.kt [hunk 5]
```diff
@@ -67,11 +66,14 @@ fun <T> MediaSmallRow(
...
+    image: String?,
+    label: String?,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
     Card(
-        modifier = Modifier
-            .wrapContentHeight()
-            .width(width)
+        modifier = modifier
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/ui/MediaSmall.kt:76]: There should be 2 widths, one for anime and one for characters

File: app/src/main/java/com/imashnake/animite/features/ui/MediaSmall.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/ui/MediaSmall.kt:55]: Yep!

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt [hunk 6]
```diff
@@ -135,177 +138,70 @@ fun Home(
...
-                                mediaList = trendingList,
-                                onItemClick = { itemId ->
+                            HomeRow(
+                                list = trendingList,
+                                title = stringResource(Res.string.trending_now),
+                                onItemClicked = {
                                     navigator.navigate(
                                         MediaPageDestination(
-                                            id = itemId,
+                                            id = it.id,
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:152]: cc @imashnake0

**Review by imashnake0 — APPROVED** [review]
LGTM. Left a couple comments but they can be addressed later since one will change after we get the models in #67; and the other is just a matter of extracting the callback which I'll leave for you to decide.

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (imashnake0) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:152]: As long as it's not hard to read (nested lambdas for instance), I don't mind.

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt [hunk 7]
```diff
@@ -345,3 +241,40 @@ fun Home(
         }
     }
 }
+
+@Composable
+fun HomeRow(
+    list: List<MediaListQuery.Medium?>,
+    title: String,
+    onItemClicked: (MediaListQuery.Medium) -> Unit,
```
[the remark this answers is not in the document]
Comment (boswelja) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:249]: Is there ever gonna be a situation where the user can click on an item that doesn't exist?
I think thefallen is working on nullability so hopefully we can get rid of double bangs with that easy enough

File: app/src/main/java/com/imashnake/animite/features/home/Home.kt [hunk 8]
```diff
@@ -135,177 +138,70 @@ fun Home(
...
-
-                            Spacer(Modifier.size(dimensionResource(Res.dimen.medium_padding)))
-
-                            MediaSmallRow(
-                                mediaList = popularList,
-                                onItemClick = { itemId ->
+                            HomeRow(
+                                list = popularList,
+                                title = stringResource(Res.string.popular_this_season),
+                                onItemClicked = {
```
[the remark this answers is not in the document]
Comment (imashnake0) [reply app/src/main/java/com/imashnake/animite/features/home/Home.kt:164]: TODO for later!

**Pull request merged**